code/modules/recycling/disposal-unit.dm

```diff
@@ -96,7 +93,21 @@
 					user << "<span class='notice'>You slice the floorweld off \the [src].</span>"
 					Deconstruct()
 				return
-	return 1
+
+	if(user.a_intent != "harm")
+		if(!user.drop_item() || (I.flags & ABSTRACT))
+			return
+		place_item_in_disposal(I, user)
+		update_icon()
+		return 1 //no afterattack
+	else
+		return ..()
+
+/obj/machinery/disposal/proc/place_item_in_disposal(obj/item/I, mob/user)
+	I.loc = src
+	user.visible_message("[user.name] places \the [I] into \the [src].", \
+						"<span class='notice'>You place \the [I] into \the [src].</span>")
```

Comment:
this proc is here to remove copypasta between disposal bin and delivery chute. It also allows the delivery chute to accept trash by simply clicking it with an item in hand, just like a disposal bin.